dynamiq/memory/backends/qdrant.py

```diff
@@ -13,46 +15,38 @@
 
 class QdrantError(Exception):
     """Base exception for Qdrant-related errors."""
-
     pass
 
 
 class Qdrant(MemoryBackend):
     """Qdrant implementation of the memory storage backend."""
 
-    name = "Qdrant"
-
-    def __init__(
-        self,
-        connection: QdrantConnection,
-        embedder: BaseEmbedder,
-        index_name: str = "conversations",
-        metric: str = "cosine",
-        on_disk: bool = False,
-        create_if_not_exist: bool = True,
-    ):
-        """Initializes the Qdrant memory storage.
-
-        Args:
-            connection: QdrantConnection instance for connecting to Qdrant
-            embedder: Embedder instance for creating embeddings
-            index_name: Name of the collection to store messages
-        """
-        self.connection = connection
-        self.index_name = index_name
-        self.embedder = embedder
-
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    name: str = "Qdrant"
+    connection: QdrantConnection
+    embedder: BaseEmbedder
+    index_name: str = Field(default="conversations")
+    metric: str = Field(default="cosine")
+    on_disk: bool = Field(default=False)
+    create_if_not_exist: bool = Field(default=True)
+    recreate_index: bool = Field(default=False)
+    vector_store: QdrantVectorStore | None = None
```

Comment:
It seems a lot of params related to `vector_store`, so don't see any sense to define them additionally (copy) in Qdrant memory store. Let's make `vector_store: QdrantVectorStore` mandatory

Reply:
The same things relate to Pinecone

Reply:
How it influence on frontend params?